src/runner_manager.py

```diff
@@ -422,3 +437,43 @@ def create_runner_info(
             )
 
         return runners
+
+    def _get_github_registration_token(self) -> str:
+        """Get token from GitHub used for registering runners.
+
+        Returns:
+            The registration token.
+        """
+        token: RegistrationToken
+        if isinstance(self.config.path, GitHubRepo):
+            token = self._clients.github.actions.create_registration_token_for_repo(
+                owner=self.config.path.owner, repo=self.config.path.repo
+            )
+        elif isinstance(self.config.path, GitHubOrg):
+            token = self._clients.github.actions.create_registration_token_for_org(
+                org=self.config.path.org
+            )
+        else:
+            assert_never(token)
```

Comment:
What does this do out of curiosity?

Reply:
It assert to the type system that this branch in the if-else statement can never be reached.

This is a new feature in Python 3.11 or typing-extensions 4.1:
https://docs.python.org/3/library/typing.html#typing.assert_never

The link includes an example usage:
https://typing.readthedocs.io/en/latest/source/unreachable.html